packages/sanity/src/core/form/studio/contexts/FormCallbacks.tsx

```diff
@@ -24,8 +24,66 @@ export const FormCallbacksContext = React.createContext<FormCallbacksValue | nul
 export const FormCallbacksProvider = memo(function FormCallbacksProvider(
   props: FormCallbacksValue & {children: React.ReactNode}
 ) {
+  const ref = useRef<FormCallbacksValue>(props)
+  ref.current = props
+
+  const transformPatches = useCallback(
+    (patches: FormPatch[]) => {
+      return ref.current?.transformPatches?.(patches) ?? []
+    },
+    [ref]
+  )
+
+  const onChange = useCallback((patchEvent: PatchEvent) => {
+    ref.current.onChange(patchEvent)
+  }, [])
+
+  const onPathFocus = useCallback((path: Path) => {
+    ref.current.onPathFocus(path)
+  }, [])
+  const onPathBlur = useCallback((path: Path) => {
+    ref.current.onPathBlur(path)
+  }, [])
+  const onPathOpen = useCallback((path: Path) => {
+    ref.current.onPathOpen(path)
+  }, [])
+  const onSetPathCollapsed = useCallback((path: Path, collapsed: boolean) => {
+    ref.current.onSetPathCollapsed(path, collapsed)
+  }, [])
+  const onSetFieldSetCollapsed = useCallback((path: Path, collapsed: boolean) => {
+    ref.current.onSetFieldSetCollapsed(path, collapsed)
+  }, [])
+  const onFieldGroupSelect = useCallback((path: Path, fieldGroupName: string) => {
+    ref.current.onFieldGroupSelect(path, fieldGroupName)
+  }, [])
+
+  const contextValue: FormCallbacksValue = useMemo(
```

Comment:
It's needed unless the context provider performs a shallow equal on the passed value (which I don't think it is?). Without the useMemo it will return a new object upon every render (even if the values stays the same)